src/gui/newwizard/states/serverurlsetupwizardstate.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) Fabian Müller <fmueller@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "serverurlsetupwizardstate.h"
+#include "determineauthtypejobfactory.h"
+#include "jobs/resolveurljobfactory.h"
+#include <QMessageBox>
+
+namespace {
+
+const QString defaultUrlSchemeC = QStringLiteral("https://");
+const QStringList supportedUrlSchemesC({ defaultUrlSchemeC, QStringLiteral("http://") });
+
+}
+
+namespace OCC::Wizard {
+
+ServerUrlSetupWizardState::ServerUrlSetupWizardState(SetupWizardContext *context)
+    : AbstractSetupWizardState(context)
+{
+    _page = new ServerUrlSetupWizardPage(_context->accountBuilder().serverUrl());
+}
+
+SetupWizardState ServerUrlSetupWizardState::state() const
+{
+    return SetupWizardState::ServerUrlState;
+}
+
+void ServerUrlSetupWizardState::evaluatePage()
+{
+    // we don't want to store any unnecessary certificates for this account when the user returns to the first page
+    // therefore we clear the certificates storage before resolving the URL
+    _context->accountBuilder() = {};
+
+    auto serverUrlSetupWizardPage = qobject_cast<ServerUrlSetupWizardPage *>(_page);
+    OC_ASSERT(serverUrlSetupWizardPage != nullptr);
+
+    const QUrl serverUrl = [serverUrlSetupWizardPage]() {
+        QString userProvidedUrl = serverUrlSetupWizardPage->userProvidedUrl();
+
+        // fix scheme if necessary
+        // using HTTPS as a default is a real ly good idea nowadays, users can still enter http:// explicitly if they wish to
+        if (!std::any_of(supportedUrlSchemesC.begin(), supportedUrlSchemesC.end(), [userProvidedUrl](const QString &scheme) {
+                return userProvidedUrl.startsWith(scheme);
+            })) {
+            qDebug() << "no URL scheme provided, prepending default URL scheme" << defaultUrlSchemeC;
```

Comment:
info or warning.

PS: Still waiting for the logging category

Reply:
It's prepared in #9801, which we should merge first.